parsec/backend/memory/vlob.py

```diff
@@ -137,60 +138,87 @@ def _get_vlob(self, organization_id, vlob_id):
             raise VlobNotFoundError(f"Vlob `{vlob_id}` doesn't exist")
 
     def _check_realm_read_access(self, organization_id, realm_id, user_id, encryption_revision):
-        can_read_roles = (
-            RealmRole.OWNER,
-            RealmRole.MANAGER,
-            RealmRole.CONTRIBUTOR,
-            RealmRole.READER,
-        )
         self._check_realm_access(
-            organization_id, realm_id, user_id, encryption_revision, can_read_roles
+            organization_id, realm_id, user_id, encryption_revision, OperationKind.DATA_READ
         )
 
     def _check_realm_write_access(self, organization_id, realm_id, user_id, encryption_revision):
-        can_write_roles = (RealmRole.OWNER, RealmRole.MANAGER, RealmRole.CONTRIBUTOR)
         self._check_realm_access(
-            organization_id, realm_id, user_id, encryption_revision, can_write_roles
+            organization_id, realm_id, user_id, encryption_revision, OperationKind.DATA_WRITE
         )
 
     def _check_realm_access(
-        self,
-        organization_id,
-        realm_id,
-        user_id,
-        encryption_revision,
-        allowed_roles,
-        expected_maintenance=False,
+        self, organization_id, realm_id, user_id, encryption_revision, operation_kind
     ):
         try:
             realm = self._realm_component._get_realm(organization_id, realm_id)
         except RealmNotFoundError:
             raise VlobNotFoundError(f"Realm `{realm_id}` doesn't exist")
 
+        # Only an owner can perform maintenance operation
+        if operation_kind == OperationKind.MAINTENANCE:
+            allowed_roles = (RealmRole.OWNER,)
+        # All roles can do read-only operation
+        elif operation_kind == OperationKind.DATA_READ:
+            allowed_roles = (
+                RealmRole.OWNER,
+                RealmRole.MANAGER,
+                RealmRole.CONTRIBUTOR,
+                RealmRole.READER,
+            )
+        # All roles except reader can do write operation
+        else:
```

Comment:
would be better to have an explicit `elif` on `OperationKind.DATA_WRITE`, then a final `else` with no allowed roles

This way the code is safe if we add new operation kind

Reply:
9ddb38c5be108a3fc89d6ff4ab2404d92401c0ed